src/App.ts

```diff
@@ -252,11 +253,50 @@ export default class App {
           signingSecret,
           endpoints,
           processBeforeResponse,
+          clientId,
+          clientSecret,
+          stateStore,
+          stateSecret,
+          installationStore,
+          authVersion,
+          metadata,
+          scopes,
           logger: this.logger,
         });
       }
     }
 
+    let usingBuiltinOauth = undefined;
+    if (
+      clientId !== undefined
+      && clientSecret !== undefined
+      && (stateSecret !== undefined || stateStore !== undefined)
+      && this.receiver instanceof ExpressReceiver
```

Comment:
let's try using a user-defined type guard to narrow the type: https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards